New rate limits documentation (#2219)
port-labs/port-docs: docs/api-reference/rate-limits.mdx

import TabItem from "@theme/TabItem";
import Export from "@theme/ApiExplorer/Export";

> # Rate Limits

Reviewer: ```suggestion
# Rate limits
```

A rate limit restricts the number of requests or actions allowed within a certain timeframe to prevent overuse or abuse.
Port enforces multiple rate limits to control extensive usage of Port's API.

Port's rate limits vary based on the API being accessed and may be more lenient for some APIs.
Port API's rate limits are enforced at the **Organization level**.

> ## Rate Limits Table

Reviewer: ```suggestion
## Rate limits table
```

| Description | Rate limit   |
|-------------|--------------|
| Unauthenticated requests | 100 requests per 5 minutes |
| Entities API requests    | 30,000 requests per 5 minutes |
| All other API requests   | 15,000 requests per 5 minutes |

> ## Checking the Rate Limit Status

Reviewer: ```suggestion
## Check the rate limit status
```

Each API response includes headers that provide information about your current rate limit status.

| Header name | Description  |
|-------------|--------------|
| x-ratelimit-period         | The time in seconds which defines the rate limit window |
| x-ratelimit-limit	         | The maximum number of requests that you can make in during the rate limit window |
| x-ratelimit-remaining      | The number of requests remaining in the current rate limit window |
| x-ratelimit-reset	         | The time at which the current rate limit window resets, in UTC epoch seconds |

> ## How to Prevent Getting Rate Limited

Reviewer: ```suggestion
## How to prevent getting rate limited
```

> ### Monitor Rate Limit Headers

Reviewer: ```suggestion
### Monitor rate limit headers
```

To prevent hitting rate limits, you should monitor the rate limit headers in your API responses.

The `x-ratelimit-remaining` header shows how many requests you have left in the current window.
When this number gets low, you should reduce your request frequency or implement a backoff strategy.

The `x-ratelimit-reset` header tells you when the rate limit window will reset, which can help you plan when to resume normal request rates.

> ### Use Exponential Backoff

Reviewer: ```suggestion
### Use exponential backoff
```

# Rate limits

> :::warning We are updating our Rate limits
> Port's rate limits will be updated on 1.06.2025
> 
> Rate limits will change from the **IP level** to the **Organization level**
> 
> <details>
> <summary>Click here to learn about the updated rate limits</summary>
> 
> The table below outlines the updated limits:
> 
> 
> | Description | Rate limit   |
> |-------------|--------------|
> | Unauthenticated requests | 100 requests per 5 minutes |
> | Entities API requests    | 30,000 requests per 5 minutes |
> | All other API requests   | 15,000 requests per 5 minutes |
> 
> 
> ## Check the rate limit status
> 
> Each API response includes headers that provide information about your current rate limit status.
> 
> | Header name | Description  |
> |-------------|--------------|
> | `x-ratelimit-period`         | The time in seconds which defines the rate limit window |
> | `x-ratelimit-limit`          | The maximum number of requests that you can make in during the rate limit window |
> | `x-ratelimit-remaining`      | The number of requests remaining in the current rate limit window |
> | `x-ratelimit-reset`          | The time at which the current rate limit window resets, in UTC epoch seconds |
> 
> 
> </details>
> :::

Reviewer: ```suggestion
:::warning Updated API rate limits
Port's API rate limits will be updated on **June 6, 2025**.

Rate limits will change from the **IP level** to the **Organization level**.

<details>
<summary>**Updated rate limits (click to expand)**</summary>

Updated limits (starting June 6, 2025):

| Description | Rate limit   |
|-------------|--------------|
| Unauthenticated requests | 100 requests per 5 minutes |
| Entities API requests    | 30,000 requests per 5 minutes |
| All other API requests   | 15,000 requests per 5 minutes |

## Check the rate limit status

Each API response includes headers that provide information about your current rate limit status:

| Header name | Description  |
|-------------|--------------|
| `x-ratelimit-period`         | The time in seconds which defines the rate limit window |
| `x-ratelimit-limit`          | The maximum number of requests that you can make in during the rate limit window |
| `x-ratelimit-remaining`      | The number of requests remaining in the current rate limit window |
| `x-ratelimit-reset`          | The time at which the current rate limit window resets, in UTC epoch seconds |

</details>

For the current rate limits, read on below.
:::
```

Author: added with some modifications

Port enforces multiple rate limits to control extensive usage of Port's API.

> Port's rate limits vary based on the API being accessed and may be more lenient for some APIs.

Reviewer: ```suggestion
Port's rate limits vary based on the API being accessed and may be more lenient for some APIs.  
```

> Port API's rate limits are enforced at the **IP level**.

Reviewer: ```suggestion
Rate limits are enforced at the **IP level**.
```

> :::note updated API rate limits starting 1.06.2025

Reviewer: ```suggestion
:::info updated API rate limits starting 1.06.2025
```

| Entity API requests | 5,000 requests per 5 minutes | 
| All other API requests | 2,500 requests per 5 minutes |

:::note updated API rate limits starting 1.06.2025

> <details>

Reviewer: lets move this "details+summary" to the first callout, no need for 2 callouts. So the first one will have the notice and the changes, and instead of saying "For more information, see the note below", let's say something like "for the current rate limits, see below"

Author: The notice should just act as a "heads-up" kind of
